File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        //BRUTE FORCE
        int count = 0;
        ListNode* temp = head;
        while(temp != nullptr){
            count++;
            temp = temp->next;
        }

        int res = count - n;

        //edge case :- if count == n; remove head

        if(count == n){
            head = head->next;
            return head;
        }

        temp = head;
        while(temp != nullptr){
            res--;
            if(res == 0){
                temp->next = temp->next->next;
                break;
            }
            temp = temp->next;
        }


        return head;

     
        
    }
};

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.dart
/**
 * Definition for singly-linked list.
 * class ListNode {
 *   int val;
 *   ListNode? next;
 *   ListNode([this.val = 0, this.next]);
 * }
 */
class Solution {
  ListNode? removeNthFromEnd(ListNode? head, int n) {
    if(head == null) return null;
    ListNode? temp = head;
    int count = 0;
    while(temp != null){
        count++;
        temp = temp.next;
    }

    int res = count - n;
    if (res == 0) {
      return head.next;
    }
    temp = head;
    for(int i  = 0 ; i < res -1 ; i++){
        if (temp != null) {
            temp = temp.next;
        }

    }
    if (temp != null && temp.next != null) temp.next = temp.next?.next;
    return head;


    
  }
}